components/x-gift-article/src/GiftArticle.jsx

```diff
@@ -41,15 +44,25 @@ const withGiftFormActions = withActions(
 			},
 
 			async createGiftUrl() {
-				const { redemptionUrl, redemptionLimit } = await api.getGiftUrl(initialProps.article.id)
-
-				if (redemptionUrl) {
-					const { url, isShortened } = await api.getShorterUrl(redemptionUrl)
-					tracking.createGiftLink(url, redemptionUrl)
+				return async (state) => {
+					let response
+					if (state.includeHighlights) {
+						const { highlightsAccessToken } = await highlightsApiClient.shareHighlights(
+							initialProps.article.id
+						)
+						response = await api.getGiftUrl(initialProps.article.id, highlightsAccessToken)
+					} else {
+						response = await api.getGiftUrl(initialProps.article.id)
+					}
+					const { redemptionUrl, redemptionLimit } = response
```

Comment:
**Suggestion:** I would if this could be simplified if `shareHighlights` accepted `state.includeHighlights` as a prop and exited early if it was falsey? That would look something like the below, what do you think?

This function

```js
const { highlightsAccessToken } = await highlightsApiClient.shareHighlights(
	initialProps.article.id
);

const { redemptionUrl, redemptionLimit } = await api.getGiftUrl(initialProps.article.id, highlightsAccessToken);

```

shareHighlights function could have this at the start

```js
async shareHighlights({ articleId, includeHighlights} = props) {
    if (!includeHighlights) {
        return {};
    }
    // rest of function
}
```

Reply:
Or another alternative would be to move the includesHighlights logic into the `getGiftUrl` function.

That function would become something like

```js
async getGiftUrl(articleId, includeHighlights = false) {
    
    let highlightsToken = undefined;
    if (includeHighlights) {
         const { highlightsAccessToken } = await highlightsApiClient.shareHighlights(
		initialProps.article.id
	);
	
	highlightsToken = highlightsAccessToken;
    }
    
    // rest of code
```

and this function would become just

```js
const { redemptionUrl, redemptionLimit } = await api.getGiftUrl(initialProps.article.id, state.includeHighlights);
```

Reply:
I think the first option is better since in the second options we will be making two api calls in one function which is not ideal